### ISSUE TITLE
Ability to Trigger CLI directly from export

### ISSUE THREAD

**Issue by ovflowd** [iamvishnusankar/next-sitemap#639]
Right now, the full workload of `next-sitemap` can only be executed by manually running the `next-sitemap` binary through `npm exec next-sitemap` or on a script via `npm run your-script-with-next-sitemap`

Allowing a direct way of executing the CLI main() action directly from a module would be great. It and also be nice, as I could leverage https://github.com/iamvishnusankar/next-sitemap/pull/638 on-demand.

The example below describes an use case:

```tsx
import { generateSitemap, generateRobots } from 'next-sitemap';
import sitemapConfig from './next-sitemap.config.mjs';

// Extra Options is any extra supported configuration option
// the `sitemap` constant is the generated XML
// instead of saving the file it gives the result
const sitemap = generateSitemap({ ...sitemapConfig, ...extraOptions });

// same for robots?
const robots = generateRobots({ ...sitemapConfig, ...extraOptions });
```

This is just an illustrative example; the config options do not necessarily need to be from the CLI config.

This is a rough draft/idea and possibly sketchy. Also, It's OK if this never gets done.

**Comment by iamvishnusankar** [comment]
@ovflowd Check this block of code. 

https://github.com/iamvishnusankar/next-sitemap/blob/6c4c35cff698c93baa021515cafeda68cf276eaf/packages/next-sitemap/src/cli.ts#L31-L47

Runtime paths interface

https://github.com/iamvishnusankar/next-sitemap/blob/6c4c35cff698c93baa021515cafeda68cf276eaf/packages/next-sitemap/src/interface.ts#L230-L239

And the chunks are `ISitemapField[][]`.

So, you should be able to generate during runtime using `ExportableBuilder`. I haven't tested it though!

**Comment by ovflowd** [comment]
Yea I saw that the builder is exported, though I would need to create a custom solution, and I wanted to stay as much as close with the official codebase, because if this gets updated and my approach isn't officially supported, then I will face issues once I need to potentially update the codebase 😅

**Comment by ovflowd** [comment]
Note this feature request is more like a "nice to have"

**Comment by iamvishnusankar** [comment]
Sure, I'll consider it for future releases. Static export support should be available from `4.1.3`

**Comment by ovflowd** [comment]
This issue got closed (?) (not really)
